console/console_test.go

```diff
@@ -0,0 +1,327 @@
+package console
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/CortexFoundation/CortexTheseus/common"
+	"github.com/CortexFoundation/CortexTheseus/consensus/cuckoo"
+	"github.com/CortexFoundation/CortexTheseus/core"
+	"github.com/CortexFoundation/CortexTheseus/ctxc"
+	"github.com/CortexFoundation/CortexTheseus/internal/jsre"
+	"github.com/CortexFoundation/CortexTheseus/miner"
+	"github.com/CortexFoundation/CortexTheseus/node"
+)
+
+const (
+	testInstance = "console-tester"
+	testAddress  = "0x8605cdbbdb6d264aa742e77020dcbc58fcdce182"
+)
+
+// hookedPrompter implements UserPrompter to simulate use input via channels.
+type hookedPrompter struct {
+	scheduler chan string
+}
+
+func (p *hookedPrompter) PromptInput(prompt string) (string, error) {
+	// Send the prompt to the tester
+	select {
+	case p.scheduler <- prompt:
+	case <-time.After(time.Second):
+		return "", errors.New("prompt timeout")
+	}
+	// Retrieve the response and feed to the console
+	select {
+	case input := <-p.scheduler:
+		return input, nil
+	case <-time.After(time.Second):
+		return "", errors.New("input timeout")
+	}
+}
+
+func (p *hookedPrompter) PromptPassword(prompt string) (string, error) {
+	return "", errors.New("not implemented")
+}
+func (p *hookedPrompter) PromptConfirm(prompt string) (bool, error) {
+	return false, errors.New("not implemented")
+}
+func (p *hookedPrompter) SetHistory(history []string)              {}
+func (p *hookedPrompter) AppendHistory(command string)             {}
+func (p *hookedPrompter) ClearHistory()                            {}
+func (p *hookedPrompter) SetWordCompleter(completer WordCompleter) {}
+
+// tester is a console test environment for the console tests to operate on.
+type tester struct {
+	workspace string
+	stack     *node.Node
+	cortex    *ctxc.Cortex
+	console   *Console
+	input     *hookedPrompter
+	output    *bytes.Buffer
+}
+
+// newTester creates a test environment based on which the console can operate.
+// Please ensure you call Close() on the returned tester to avoid leaks.
+func newTester(t *testing.T, confOverride func(*ctxc.Config)) *tester {
+	// Create a temporary storage for the node keys and initialize it
+	workspace, err := ioutil.TempDir("", "console-tester-")
+	if err != nil {
+		t.Fatalf("failed to create temporary keystore: %v", err)
+	}
+
+	// Create a networkless protocol stack and start an Cortex service within
+	stack, err := node.New(&node.Config{DataDir: workspace, UseLightweightKDF: true, Name: testInstance})
+	if err != nil {
+		t.Fatalf("failed to create node: %v", err)
+	}
+	ctxcConf := &ctxc.Config{
+		//TODO:change to devloper
+		Genesis: core.DefaultGenesisBlock(),
+		Miner: miner.Config{
```

Comment:
add miner.coinbase here,but can't get it by ctxc.coinbase.

Reply:
```
func (miner *Miner) SetCoinbase(addr common.Address) {
        miner.coinbase = addr
        miner.worker.setCoinbase(addr)
}
```

Reply:
coinbase can be only set by SetCoinbase

Reply:
Etherbase = Coinbase

Reply:
But I've already set it up through ctxcConf.miner.coinbase

Reply:
> But I've already set it up through ctxcConf.miner.coinbase，doesn't that work?

Also right